src/components/molecules/Dialog/hook/useDialog.tsx

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useState } from 'react'
+import { Dialog as Component, DialogProps } from '@/components/molecules/Dialog'
+
+export type UseDialogReturnType = {
+  isOpen: boolean
+  open: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  close: () => void
+  Dialog: React.FC<DialogProps>
+}
+
+type ComponentType = Pick<DialogProps, 'title' | 'children'>
+
+export const useDialog = (): UseDialogReturnType => {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+
+  const open = useCallback(
```

Comment:
```suggestion
  const open: UseDialogReturnType["open"] = useCallback(
```

こっちの方が型の管理は楽かなと思います。